app/src/gecko/java/org/mozilla/tv/firefox/webrender/WebRenderComponents.kt

```diff
@@ -25,6 +28,19 @@ class WebRenderComponents(applicationContext: Context, systemUserAgent: String)
                 systemUserAgent = systemUserAgent,
                 appName = applicationContext.resources.getString(R.string.useragent_appname))
 
+        val runtimeSettingsBuilder = GeckoRuntimeSettings.Builder()
+        if (BuildConstants.isDevBuild) {
+            // In debug builds, allow to invoke via an Intent that has extras customizing Gecko.
+            // In particular, this allows to add command line arguments for custom profiles, etc.
+            val extras = applicationContext.serviceLocator.launchSafeIntent.extras
```

Comment:
Is this copy by reference? If so, shouldn't this be synchronized? But I guess lazy{} by default grants thread safe. This seems flaky since `Engine` Initialization happens prior to `Activity.onCreate()`. Also closing the app does not guarantee the Activity to be destroyed

Reply:
I'm not sure what the ownership model is here.  The `launchSafeIntent` itself is synchronized by the Swift runtime, and I expect all access to `extras` to be read-only, but that's (probably?) not guaranteed.

Re: flakiness and initialization order, that's the devil we must dance with.  Gecko is initialized at most once *per process*, and it's initialized very early.  I am not aware of a way to get the launching Intent extras (if there are any) before the first `Activity.onCreate()`.  So yes, this necessarily races Gecko startup.  @psymoon, do you know of such a way to get the extras in another place?  It can't be in `Application.onCreate` since a service could start the App before any Activity starts GeckoView...

Reply:
Also, it cannot be true that `Engine` initialization happens prior to `Activity.onCreate`, because locally the flags I send in as Intent extras make it to the GV creation.

Reply:
> Also, it cannot be true that `Engine` initialization happens prior to `Activity.onCreate`, because locally the flags I send in as Intent extras make it to the GV creation.

I think you're right technically. If it's not, we are doing something wrong...

> @psymoon, do you know of such a way to get the extras in another place? It can't be in Application.onCreate since a service could start the App before any Activity starts GeckoView...

Activity.onCreate is the correct way, I think since we are checking for null anyway it may be just fine